### DISCUSSION TITLE
Tips for debugging `custom/chadrc.lua`

### DISCUSSION THREAD

**Discussion by erwin** [NvChad/NvChad#1589] [General]
UPDATE:

Figured out a lazier way to do this, in your custom/nvchad.lua add:

```lua
vim.cmd([[command! NVT lua print(vim.inspect(require('custom/chadrc')))]])
```

Then inside of neovim just run `:NVT` to syntax check

And if you have any unexpected problems, run it from the command line:

```shell
nvim -u NONE --cmd "lua print(vim.inspect(require('custom/chadrc')))"
```

And this way, easy to add a shell alias so you don't have to remember this.

---

Frequently I make changes to my `chadrc` or other custom config files that end up breaking my config and then everything just silently fails by default.

Today, I found a bit of a process that at least was helpful for me.

Perhaps more useful, you can run:

```
nvim -u NONE
```

Then, inside neovim run:

```lua
:lua M=require('custom/chadrc')
:lua print(vim.inspect(M))
```

or

```
:lua print(vim.inspect(require('custom/chadrc')))
```

If you've got a syntax error, you should get a clue about where it is like this:
![image](https://user-images.githubusercontent.com/11534217/191939001-49eb9f8d-d906-4d32-8631-1a91cf5c6136.png)

If you don't have any syntax errors, you should get a pretty print of the table defined by chadrc.lua:

![image](https://user-images.githubusercontent.com/11534217/191939252-94420fe4-b71e-4a5e-9423-bdbaf8d2d6da.png)

If your chadrc.lua is correct (no errors) this should probably work for you in  a regular neovim session, but when I hit **any** problems, I like to drop back to `-u NONE` right away. Simplicity clears my mind and helps me focus.

If you're unable to require your chadrc, there's probably a syntax error that you might check with on the command line with a Lua runtime, like `lua` or `luajit`.

```shell
lua chadrc.lua
```

It will almost always report errors of some type, because the vim/nvim tables that your config probably depends on won't exist, but it's another way to help find syntax errors that prevent your `chadrc.lua` from running.

Hopefully that's helpful to you and not a waste of your time!

**Comment by siduck** [comment]
@erwin thx for this! will not be useful for me as i just write correct code & syntax but for newbies it'd be very helpful!
